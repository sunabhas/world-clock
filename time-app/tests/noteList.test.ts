import { mount } from "@vue/test-utils";
import NoteList from "../components/NotesList.vue";

describe("NoteList", () => {
  it('should emit "handlePrevious" when previous button is clicked', async () => {
    const wrapper = mount(NoteList, {
      props: {
        showPagination: true,
        previousDisabled: false,
        nextDisabled: false,
      },
    });
    const prevButton = wrapper.find('[data-testid="prev-button"]');
    expect(prevButton).toBeDefined();
  });

  it('should emit "handleNext" when next button is clicked', async () => {
    const wrapper = mount(NoteList, {
      props: {
        showPagination: true,
        previousDisabled: false,
        nextDisabled: false,
      },
    });

    const nextButton = wrapper.find('[data-testid="next-button"]');
    expect(nextButton).toBeDefined();
  });
});
