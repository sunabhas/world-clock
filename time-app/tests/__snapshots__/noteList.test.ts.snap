// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoteList > renders with given props 1`] = `
<div>
  <div
    class="p-2 my-4"
  >
    <custom-notes-table
      tabledata="[object Object]"
      tableheaders="Note,Time,Area,Location"
    />
  </div>
</div>
`;
